cloud/deployment/deployment_variable.go

```diff
@@ -218,6 +221,54 @@ func addVariableFromFlag(oldKeyList []string, oldEnvironmentVariables []astro.En
 	return newEnvironmentVariables
 }
 
+func addVariablesFromArgs(oldKeyList []string, oldEnvironmentVariables []astro.EnvironmentVariablesObject, newEnvironmentVariables []astro.EnvironmentVariable, variableList []string, updateVars, makeSecret bool) []astro.EnvironmentVariable {
+	var key string
+	var val string
+	// valisdate each key value pair and add to new variables list
+	for i := range variableList {
+		// split pair
+		pair := strings.Split(variableList[i], "=")
+		if len(pair) > 1 {
+			key = pair[0]
+			val = pair[1]
+			if key == "" || val == "" {
+				fmt.Printf("Input %s has blank key or value\n", variableList[i])
+				continue
+			}
+		} else {
+			fmt.Printf("Input %s is not a valid key value pair\n", variableList[i])
+			continue
+		}
+		var newEnvironmentVariable astro.EnvironmentVariable
+		exist, num := contains(oldKeyList, key)
+		switch {
+		case exist && !updateVars: // don't update variable
+			fmt.Printf("key %s already exists, skipping creation. Use the update command to update existing variables", key)
+		case exist && updateVars: // update variable
+			fmt.Printf("updating variable %s \n", key)
+			secret := makeSecret
+			if !makeSecret { // you can only make variables secret a user can't make them not secret
+				secret = oldEnvironmentVariables[num].IsSecret
+			}
+			newEnvironmentVariable = astro.EnvironmentVariable{
+				IsSecret: secret,
+				Key:      oldEnvironmentVariables[num].Key,
+				Value:    val,
+			}
+			newEnvironmentVariables[num] = newEnvironmentVariable
+		default:
+			newFileEnvironmentVariable := astro.EnvironmentVariable{
+				IsSecret: makeSecret,
+				Key:      key,
+				Value:    val,
+			}
+			newEnvironmentVariables = append(newEnvironmentVariables, newFileEnvironmentVariable)
+			fmt.Printf("adding variable %s\n", key)
+		}
```

Comment:
should this logic be better in a separate function, since the same logic is being used in multiple places now?
Your call, you can do it in a separate PR as well.